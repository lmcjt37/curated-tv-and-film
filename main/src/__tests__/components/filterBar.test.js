import React from 'react';
import ReactDOM from 'react-dom';

import FilterBar from '../../components/filterBar';

import { createRender } from '@material-ui/core/test-utils';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const yearsMock = [2017, 2018, 2019, 2020];
const mockFunction = jest.fn();

var state = {
  years: yearsMock,
  filterResults: 'all',
  filterYear: 'All',
  filterGenre: 'All',
  filterOrder: 'Ascending'
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<FilterBar {...state} handleFilter={mockFunction} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('snapshot of initial component', () => {
  let render = createRender();
  const component = render(
    <FilterBar {...state} handleFilter={mockFunction} />
  );
  expect(component).toMatchSnapshot();
});
