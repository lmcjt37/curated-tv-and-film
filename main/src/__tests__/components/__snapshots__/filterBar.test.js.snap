// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of initial component 1`] = `
<div
  className="filter-bar"
>
  <div
    className="filter-bar__container"
  >
    <input
      checked={false}
      name="filterResults"
      onChange={[Function]}
      type="radio"
      value="all"
    />
    <label
      className="filter-labels"
      htmlFor="filterAll"
    >
      All
    </label>
    <input
      checked={false}
      name="filterResults"
      onChange={[Function]}
      type="radio"
      value="movies"
    />
    <label
      className="filter-labels"
      htmlFor="filterMovies"
    >
      Movies
    </label>
    <input
      checked={false}
      name="filterResults"
      onChange={[Function]}
      type="radio"
      value="tv"
    />
    <label
      className="filter-labels"
      htmlFor="filterTV"
    >
      TV Series
    </label>
    <label
      className="filter-labels"
      htmlFor="filterGenre"
    >
      Genre
    </label>
    <select
      name="filterGenre"
      type="select"
    >
      <option
        value="All"
      >
        All
      </option>
      <option
        value="Action"
      >
        Action
      </option>
      <option
        value="Adventure"
      >
        Adventure
      </option>
      <option
        value="Animation"
      >
        Animation
      </option>
      <option
        value="Comedy"
      >
        Comedy
      </option>
      <option
        value="Crime"
      >
        Crime
      </option>
      <option
        value="Drama"
      >
        Drama
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Mystery"
      >
        Mystery
      </option>
      <option
        value="Romance"
      >
        Romance
      </option>
      <option
        value="Sci-Fi"
      >
        Sci-Fi
      </option>
      <option
        value="Thriller"
      >
        Thriller
      </option>
      <option
        value="Horror"
      >
        Horror
      </option>
    </select>
    <label
      className="filter-labels"
      htmlFor="filterAlpha"
    >
      Alphabetical Order
    </label>
    <select
      name="filterAlpha"
      type="select"
    >
      <option
        value="Ascending"
      >
        Ascending
      </option>
      <option
        value="Descending"
      >
        Descending
      </option>
    </select>
  </div>
</div>
`;
